sympy/geometry/util.py

```diff
@@ -570,12 +571,22 @@ def idiff(eq, y, x, n=1):
         y = y[0]
     elif isinstance(y, Symbol):
         dep = {y}
+    elif isinstance(y, Function):
+        pass
     else:
-        raise ValueError("expecting x-dependent symbol(s) but got: %s" % y)
+        raise ValueError("expecting x-dependent symbol(s) or function(s) but got: %s" % y)
+
+    if isinstance(eq, Equality):
```

Comment:
oops! I missed this. A derivative of a bool is something that might be defined (see [here](https://github.com/sympy/sympy/issues/13350)) but if you use the `Eq` as an equation container in this routine, it will never be able to calculate a derivative of this boolean. If the *user* wants to use `Eq` as an equation container, that is their choice; if they want to use `idiff` *they* will have to pass it as `eq.rewrite(Add)`; this is not something that `idiff` should do.